modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionImpl.java

```diff
@@ -330,7 +329,13 @@ public Publisher<Long> executeBatchReactive(@Nullable Transaction transaction, S
     /** {@inheritDoc} */
     @Override
     public void close() {
-        await(closeAsync());
+        try {
+            closeAsync().toCompletableFuture().get();
+        } catch (ExecutionException e) {
+            throw new SqlException(RUNTIME_EXECUTION_ERR, e.getCause());
```

Comment:
SqlException may rewrite code from cause.
Is it done intentionally?

Reply:
The code behaves exactly as it has been prior to this change.

Reply:
Fixed.